sdk/communication/azure-communication-phonenumbers/src/main/java/com/azure/communication/phonenumbers/SipRoutingClientBuilder.java

```diff
@@ -0,0 +1,451 @@
+import com.azure.core.http.HttpPipeline;
+import com.azure.core.http.HttpPipelineBuilder;
+import com.azure.core.http.policy.BearerTokenAuthenticationPolicy;
+import com.azure.core.http.policy.CookiePolicy;
+import com.azure.core.http.policy.HttpLogDetailLevel;
+import com.azure.core.http.policy.HttpLogOptions;
+import com.azure.core.http.policy.HttpLoggingPolicy;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import com.azure.core.http.policy.RequestIdPolicy;
+import com.azure.core.http.policy.RetryOptions;
+import com.azure.core.http.policy.RetryPolicy;
+import com.azure.core.http.policy.UserAgentPolicy;
+import com.azure.core.util.ClientOptions;
+import com.azure.core.util.Configuration;
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.HttpClientOptions;
+import com.azure.core.util.builder.ClientBuilderUtil;
+import com.azure.core.util.logging.ClientLogger;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Builder for creating clients of Communication Service SIP routing configuration
+ */
+@ServiceClientBuilder(serviceClients = {SipRoutingClient.class, SipRoutingAsyncClient.class})
+public final class SipRoutingClientBuilder implements
+    AzureKeyCredentialTrait<SipRoutingClientBuilder>,
+    ConfigurationTrait<SipRoutingClientBuilder>,
+    ConnectionStringTrait<SipRoutingClientBuilder>,
+    EndpointTrait<SipRoutingClientBuilder>,
+    HttpTrait<SipRoutingClientBuilder>,
+    TokenCredentialTrait<SipRoutingClientBuilder> {
+    private static final String APP_CONFIG_PROPERTIES = "azure-communication-phonenumbers-siprouting.properties";
+    private static final Map<String, String> PROPERTIES = CoreUtils.getProperties(APP_CONFIG_PROPERTIES);
+    private static final String SDK_NAME = "name";
+    private static final String SDK_VERSION = "version";
+
+    private final ClientLogger logger = new ClientLogger(SipRoutingClientBuilder.class);
+
+    private SipRoutingServiceVersion version = SipRoutingServiceVersion.getLatest();
+    private String endpoint;
+    private HttpPipeline pipeline;
+    private HttpClient httpClient;
+    private HttpLogOptions httpLogOptions;
+    private AzureKeyCredential azureKeyCredential;
+    private TokenCredential tokenCredential;
+    private Configuration configuration;
+    private ClientOptions clientOptions;
+    private RetryPolicy retryPolicy;
+    private RetryOptions retryOptions;
+    private final List<HttpPipelinePolicy> additionalPolicies = new ArrayList<>();
+
+    /**
+     * Sets endpoint of the service
+     *
+     * @param endpoint url of the service
+     * @return The updated {@link SipRoutingClientBuilder} object.
+     * @throws NullPointerException If {@code endpoint} is {@code null}.
+     */
+    @Override
+    public SipRoutingClientBuilder endpoint(String endpoint) {
+        this.endpoint = Objects.requireNonNull(endpoint, "'endpoint' cannot be null.");
```

Comment:
Should also check that the endpoint is valid in the setter

[Example of verifying endpoint](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/SearchClientBuilder.java#L215)

Reply:
Check implemented.